source/Ship.cpp

```diff
@@ -234,8 +235,18 @@ void Ship::Load(const DataNode &node)
 					? "no key." : "key: " + child.Token(1)));
 			continue;
 		}
-		if(key == "sprite")
-			LoadSprite(child);
+		if(key == "sprite" || key == "sprite-flying")
+			LoadSprite(child, Body::BodyState::FLYING);
+		else if(key == "sprite-firing")
+			LoadSprite(child, Body::BodyState::FIRING);
+		else if(key == "sprite-landing")
+			LoadSprite(child, Body::BodyState::LANDING);
+		else if(key == "sprite-launching")
+			LoadSprite(child, Body::BodyState::LAUNCHING);
+		else if(key == "sprite-jumping")
+			LoadSprite(child, Body::BodyState::JUMPING);
+		else if(key == "sprite-disabled")
```

Comment:
Just a thougt,because it would be more elegant but you dont have to: what if we define these in a map and then look if the element is in the map, and automatically assign it to the right bodystate?

Reply:
Good idea, will make that more concise.